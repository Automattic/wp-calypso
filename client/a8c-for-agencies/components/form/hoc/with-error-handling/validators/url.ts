import { translate } from 'i18n-calypso';
import { isValidUrl } from 'calypso/a8c-for-agencies/sections/partner-directory/utils/tools';
import { FieldTypes } from '../types';

const validateUrl = ( message?: string ) => ( field?: FieldTypes ) => {
	if ( ! field || ! field?.length ) {
		return null;
	}

	if ( typeof field !== 'string' || ! isValidUrl( field ) ) {
		return message ?? translate( 'Please provide correct URL' );
	}
	return null;
};

export default validateUrl;
