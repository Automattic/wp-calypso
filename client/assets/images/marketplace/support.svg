<svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
<mask id="mask0_7786_53482" style="mask-type:alpha" maskUnits="userSpaceOnUse" x="2" y="0" width="20" height="24">
<path fill-rule="evenodd" clip-rule="evenodd" d="M11.5 19.695V23.295L12.93 22.605C18.06 20.135 21.5 15.155 21.5 10.205C21.5 4.96499 17.24 0.704987 12 0.704987C6.76 0.704987 2.5 4.96499 2.5 10.205C2.5 15.275 6.49 19.435 11.5 19.695ZM4.5 10.205C4.5 6.06499 7.86 2.70499 12 2.70499C16.14 2.70499 19.5 6.06499 19.5 10.205C19.5 13.925 17.14 17.705 13.5 20.005V17.705H12C7.86 17.705 4.5 14.345 4.5 10.205ZM13 14.205V16.205H11V14.205H13ZM12.7704 9.52666C11.9421 10.1616 11 10.8837 11 12.705H13C13 11.6096 13.711 10.9883 14.4408 10.3505C15.21 9.67831 16 8.9879 16 7.70499C16 5.49499 14.21 3.70499 12 3.70499C9.79 3.70499 8 5.49499 8 7.70499H10C10 6.60499 10.9 5.70499 12 5.70499C13.1 5.70499 14 6.60499 14 7.70499C14 8.5842 13.4202 9.02859 12.7704 9.52666Z" fill="white"/>
</mask>
<g mask="url(#mask0_7786_53482)">
<rect width="24" height="24" fill="black"/>
</g>
</svg>
