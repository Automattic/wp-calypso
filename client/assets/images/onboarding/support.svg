<svg width="20" height="18" viewBox="0 0 20 18" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M9 18V16.5H17.5V8.9C17.5 7.93333 17.2917 7.00417 16.875 6.1125C16.4583 5.22083 15.9 4.43333 15.2 3.75C14.5 3.06667 13.7 2.52083 12.8 2.1125C11.9 1.70417 10.9667 1.5 10 1.5C9.03333 1.5 8.1 1.70417 7.2 2.1125C6.3 2.52083 5.5 3.06667 4.8 3.75C4.1 4.43333 3.54167 5.22083 3.125 6.1125C2.70833 7.00417 2.5 7.93333 2.5 8.9V15H2C1.45 15 0.979167 14.8042 0.5875 14.4125C0.195833 14.0208 0 13.55 0 13V11C0 10.6167 0.0916668 10.2792 0.275 9.9875C0.458333 9.69583 0.7 9.45833 1 9.275L1.075 7.95C1.225 6.73333 1.57083 5.63333 2.1125 4.65C2.65417 3.66667 3.32917 2.83333 4.1375 2.15C4.94583 1.46667 5.85417 0.9375 6.8625 0.5625C7.87083 0.1875 8.91667 0 10 0C11.1 0 12.1542 0.1875 13.1625 0.5625C14.1708 0.9375 15.075 1.47083 15.875 2.1625C16.675 2.85417 17.3458 3.6875 17.8875 4.6625C18.4292 5.6375 18.775 6.725 18.925 7.925L19 9.225C19.3 9.375 19.5417 9.59583 19.725 9.8875C19.9083 10.1792 20 10.5 20 10.85V13.15C20 13.5167 19.9083 13.8417 19.725 14.125C19.5417 14.4083 19.3 14.625 19 14.775V16.5C19 16.9167 18.8542 17.2708 18.5625 17.5625C18.2708 17.8542 17.9167 18 17.5 18H9ZM7 10.75C6.8 10.75 6.625 10.675 6.475 10.525C6.325 10.375 6.25 10.1917 6.25 9.975C6.25 9.775 6.325 9.60417 6.475 9.4625C6.625 9.32083 6.80833 9.25 7.025 9.25C7.225 9.25 7.39583 9.32083 7.5375 9.4625C7.67917 9.60417 7.75 9.78333 7.75 10C7.75 10.2 7.67917 10.375 7.5375 10.525C7.39583 10.675 7.21667 10.75 7 10.75ZM13 10.75C12.8 10.75 12.625 10.675 12.475 10.525C12.325 10.375 12.25 10.1917 12.25 9.975C12.25 9.775 12.325 9.60417 12.475 9.4625C12.625 9.32083 12.8083 9.25 13.025 9.25C13.225 9.25 13.3958 9.32083 13.5375 9.4625C13.6792 9.60417 13.75 9.78333 13.75 10C13.75 10.2 13.6792 10.375 13.5375 10.525C13.3958 10.675 13.2167 10.75 13 10.75ZM4.025 9.45C3.95833 8.46667 4.09583 7.575 4.4375 6.775C4.77917 5.975 5.2375 5.29583 5.8125 4.7375C6.3875 4.17917 7.05 3.75 7.8 3.45C8.55 3.15 9.3 3 10.05 3C11.5667 3 12.8417 3.47917 13.875 4.4375C14.9083 5.39583 15.5417 6.59167 15.775 8.025C14.2083 8.00833 12.8292 7.5875 11.6375 6.7625C10.4458 5.9375 9.525 4.86667 8.875 3.55C8.60833 4.9 8.04583 6.09583 7.1875 7.1375C6.32917 8.17917 5.275 8.95 4.025 9.45Z" fill="#007CBA"/>
</svg>
