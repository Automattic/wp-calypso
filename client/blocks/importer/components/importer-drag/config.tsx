import { translate, TranslateOptions, TranslateOptionsText } from 'i18n-calypso';
import React from 'react';
import { convertPlatformName } from 'calypso/blocks/import/util';
import importerConfig, { ImporterConfig } from 'calypso/lib/importer/importer-config';
import SupportLink from '../support-link';
import type { Importer } from 'calypso/blocks/importer/types';

const translateConfig: { [ key: string ]: Partial< ImporterConfig > } = {
	blogger: {
		description:
			'Import posts, pages, comments, tags, and images from a %(importerName)s export file.',
		uploadDescription:
			'A %(importerName)s export is ' +
			'an XML file with your page and post content, or a zip archive ' +
			'containing several XML files. ' +
			'{{supportLink/}}',
	},
	medium: {
		description:
			'Import your posts, tags, images, and videos from your %(importerName)s export file',
		uploadDescription:
			'A %(importerName)s export file is a ZIP file containing several HTML files with your stories. ' +
			'{{supportLink/}}',
	},
	squarespace: {
		description:
			'Import posts, pages, comments, tags, and images from a %(importerName)s export file.',
		uploadDescription:
			'A %(importerName)s export is ' +
			'an XML file with your page and post content, or a zip archive ' +
			'containing several XML files. ' +
			'{{supportLink/}}',
	},
	wordpress: {
		description: 'Import posts, pages, and media from your %(importerName)s export file.',
		uploadDescription:
			'A %(importerName)s export is ' +
			'an XML file with your page and post content, or a zip archive ' +
			'containing several XML files. ' +
			'{{supportLink/}}',
	},
};

export function getImportDragConfig( importer: Importer, supportLinkModal?: boolean ) {
	const importerData = importerConfig()[ importer ];

	const options: TranslateOptions = {
		args: {
			importerName: convertPlatformName( importer ),
		},
		components: {
			supportLink: <SupportLink importer={ importer } supportLinkModal={ supportLinkModal } />,
		},
	};

	importerData.title = translate( 'Import content from %(importerName)s', {
		...options,
		textOnly: true,
	} as TranslateOptionsText ) as string;

	importerData.description = translate(
		// eslint-disable-next-line wpcalypso/i18n-no-variables
		translateConfig[ importer ].description as string,
		options
	);

	importerData.uploadDescription = translate(
		// eslint-disable-next-line wpcalypso/i18n-no-variables
		translateConfig[ importer ].uploadDescription as string,
		options
	);

	return importerData;
}
