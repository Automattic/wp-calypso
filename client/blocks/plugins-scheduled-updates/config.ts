export const MAX_SELECTABLE_PLUGINS = 10;
export const MAX_SELECTABLE_PATHS = 5;
