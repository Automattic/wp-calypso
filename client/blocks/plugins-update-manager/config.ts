export const MAX_SCHEDULES = 2;
export const MAX_SELECTABLE_PLUGINS = 10;
