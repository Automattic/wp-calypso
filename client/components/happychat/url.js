/** @format */

/**
 * External dependencies
 */
import { startsWith } from 'lodash';

const schemeRegex = /^\w+:\/\//;

export const addSchemeIfMissing = ( url, scheme ) => {
	if ( false === schemeRegex.test( url ) ) {
		return scheme + '://' + url;
	}
	return url;
};

export const setUrlScheme = ( url, scheme ) => {
	const schemeWithSlashes = scheme + '://';
	if ( startsWith( url, schemeWithSlashes ) ) {
		return url;
	}

	const newUrl = addSchemeIfMissing( url, scheme );
	if ( newUrl !== url ) {
		return newUrl;
	}

	return url.replace( schemeRegex, schemeWithSlashes );
};
