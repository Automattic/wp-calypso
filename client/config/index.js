/**
 * External dependencies
 */
import cookie from 'cookie';

/**
 * Internal dependencies
 */

import createConfig from 'calypso/lib/create-config';

/**
 * Manages config flags for various deployment builds
 *
 * @module config/index
 */
if ( 'undefined' === typeof window || ! window.configData ) {
	throw new ReferenceError(
		'No configuration was found: please see client/config/README.md for more information'
	);
}

const configData = window.configData;

// calypso.live matches
// hash-abcd1234.calypso.live matches
// calypso.live.com doesn't match
const CALYPSO_LIVE_REGEX = /^([a-zA-Z0-9-]+\.)?calypso\.live$/;

// check if the current browser location is *.calypso.live
export function isCalypsoLive() {
	return typeof window !== 'undefined' && CALYPSO_LIVE_REGEX.test( window.location.host );
}

function applyFlags( flagsString, modificationMethod ) {
	const flags = flagsString.split( ',' );
	flags.forEach( ( flagRaw ) => {
		const flag = flagRaw.replace( /^[-+]/, '' );
		const enabled = ! /^-/.test( flagRaw );
		configData.features[ flag ] = enabled;
		// eslint-disable-next-line no-console
		console.log(
			'%cConfig flag %s via %s: %s',
			'font-weight: bold;',
			enabled ? 'enabled' : 'disabled',
			modificationMethod,
			flag
		);
	} );
}

const flagEnvironments = [ 'wpcalypso', 'horizon', 'stage', 'jetpack-cloud-stage' ];

if (
	process.env.NODE_ENV === 'development' ||
	flagEnvironments.includes( configData.env_id ) ||
	isCalypsoLive()
) {
	const cookies = cookie.parse( document.cookie );

	if ( cookies.flags ) {
		applyFlags( cookies.flags, 'cookie' );
	}

	const match =
		document.location.search && document.location.search.match( /[?&]flags=([^&]+)(&|$)/ );
	if ( match ) {
		applyFlags( match[ 1 ], 'URL' );
	}
}

const configApi = createConfig( configData );

export default configApi;
export const isEnabled = configApi.isEnabled;
