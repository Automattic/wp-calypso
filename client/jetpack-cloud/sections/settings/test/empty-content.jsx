/**
 * @jest-environment jsdom
 */

/**
 * External dependencies
 */
import React from 'react';
import '@testing-library/jest-dom/extend-expect';

/**
 * Internal dependencies
 */
import { JETPACK_PRICING_PAGE } from 'calypso/lib/url/support';
import { render } from 'config/testing-library';
import NoSitePurchasesMessage from '../empty-content';

describe( 'NoSitePurchasesMessage', () => {
	it( 'should render correctly', () => {
		const { container } = render( <NoSitePurchasesMessage /> );

		expect( container ).toMatchSnapshot();
	} );

	it( 'should link to the pricing page', () => {
		const { container } = render( <NoSitePurchasesMessage /> );

		expect( container.querySelector( 'a' ).getAttribute( 'href' ) ).toEqual( JETPACK_PRICING_PAGE );
	} );
} );
