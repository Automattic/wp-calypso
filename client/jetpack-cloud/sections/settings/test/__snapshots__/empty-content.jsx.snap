// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoSitePurchasesMessage should render correctly 1`] = `
<div>
  <main
    class="empty-content main"
    role="main"
  >
    <div
      class="empty-content"
    >
      <img
        alt=""
        class="empty-content__illustration"
        src="/calypso/images/illustrations/security.svg"
        width="200"
      />
      <h2
        class="empty-content__title"
      >
        Upgrade for quick restores
      </h2>
      <h3
        class="empty-content__line"
      >
        Server credentials are used to restore your site and fix any vulnerabilities
      </h3>
      <a
        class="empty-content__action button is-primary"
        href="https://jetpack.com/pricing/"
      >
        Upgrade for quick restores
      </a>
    </div>
  </main>
</div>
`;
